<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">

<html>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta name="keywords" content="Roslin Institute">
<meta name="description" content="">
<title>SIRE-PC</title>

<head>
<style>

div.box{
 border-radius: 25px;
    background: #ddddff;
    padding: 5px;
    width: 200px;
    height: 150px; 
}

div.boxgreen{
 border-radius: 25px;
    background: #ddffdd;
    padding: 5px;
    width: 200px;
    height: 150px; 
}

div.boxred{
 border-radius: 25px;
    background: #ffdddd;
    padding: 5px;
    width: 200px;
    height: 150px; 
}

div.box2{
 border-radius: 20px;
    background: #ccccff;
    padding: 10px;
    width: 200px;
    height: 150px; 
}

div.shad{
   position: fixed;
   top: 0; bottom: 0; left: 0; right: 0;
}	

div.inp{
 position: fixed;
     margin: auto;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
    width: 250px;
    height: 150px;
	}
	
div.help{
  position: fixed;
  margin: auto;
  top: 0;
  right: 5%;
  bottom: 0;
    width: 50%;
    height: 80%;
	}
	
div.paramdef{
  position: fixed;
  margin: auto;
  top: 0;
  right: 5%;
  bottom: 0;
    width: 47%;
    height: 80%;
	}
	
input.br{
   position:absolute;
    margin: auto;
    bottom:10px;
    right:10px;
	}
	
div.close{
   position:absolute;
    margin: auto;
    top:10px;
    right:0px;
}


</style>

<script>
var dx = 220, mdx = 25, dy = 170, mdy = 20, px1 = mdx, py1 = dy-mdy, px2 = dx-mdx, py2 = dy-mdy, px3 = dx/2, py3 = mdy;

var oldtype; 
var model, modelold;

var ntrial = 0, NI, NS, dom;
var triallist=[], phiBBseedlist=[], phiABseedlist=[], phiBBcontlist=[], phiABcontlist=[];
var xiI=[], HI=[], xiS=[], HS=[], etaS=[], etaI=[];

var butdown = 0, mx, my, overN=-1, overty=0;

function addtrial()
{
	num = Math.random();
	na = "trial"+num;

	triallist.push(na); ntrial++;
	
	phiBBseedlist.push(0.5);
	phiABseedlist.push(0);
	phiBBcontlist.push(0.5);
	phiABcontlist.push(0);
	
	var node = document.createElement(na);   

	var h1 = 415, h2 = 170;
	if(model == "VAC"){ h1 = 220; h2 = 70;}
	
	st = "";
	st += "<div class='box' id='" + na + "' style='float:left; width:240px; height:"+h1+"px; margin:8px; font-size:17px;'>";
	if(newtype=="User defined"){
		st += "<div style='float:right; margin-right:10px; color:red'><a style='color:red; text-decoration:none;' href='javascript:t = 0; while(t < ntrial && triallist[t] != \""+na+"\") t++; removetrial(t);'>✖</a></div>";
	}
	
	st += "<center><b style='font-size:20px' id='num"+na+"'>Group "+ntrial+"</b></center>";
	
	st += "<div class='box2' style='width:220px; height:"+h2+"px'>";
	st += add_vac_box(0,na,num);
	st += "</div>";

	st += "<div class='box2' style='width:220px; height:"+h2+"px; margin-top:10px'>";
	st += add_vac_box(1,na,num);
	st += "</div>";

	
	st += "</div>";
	node.innerHTML = st;
	
	ById("list").appendChild(node);  
}

function add_vac_box(type,na,num)
{
	var typest = "seed"; if(type == 1) typest = "cont";

	var st = "";
	if(model == "VAC"){ 
		st += "<span style = 'font:italic 16px Arial'>";
		if(type == 0) st += "Seeders:";
		else st += "Contacts:";
		st += "</span>";
		
		var N;
		if(type == 0) N = parseInt(ById("nseeder").value);
		else N = parseInt(ById("ncontact").value);

		st += "<input id='vac_slider_"+typest+"_"+na+"' type='range' style='width:210px; height:20px; margin:0px; padding:0px; margin-left:5px;  margin-right:5px;' min='0' max = '1' step='0.001' onmouseup='recalcphiBB(); calc();' oninput='recalcphiBB(); calc()' value='0'>";
		st += "<br>"; 
		st += "<div style='margin-top:10px' id='"+typest+"_"+na+"'></div>";
	}
	else{
		st += "<canvas onmousemove='move(event,"+num+","+type+")' onclick='if(overN >= 0) inputon();' onmousedown='butdown=1; move(event,"+num+","+type+");' onmouseup='butdown=0;' id='"+typest+"_"+na+"' width='"+dx+"' height='"+dy+"'></canvas>";
	}	
		
	return st;
}


function recalcphiBB()
{
	for(var t = 0; t < ntrial; t++){
		phiBBseedlist[t] = ById("vac_slider_seed_"+triallist[t]).value;
		phiBBcontlist[t] = ById("vac_slider_cont_"+triallist[t]).value;
	}	
	replot();
}
	
function removetrial(t)
{
	if(ntrial <= 1){ alert("There must be at least one contact group!"); return;}
	ById("list").removeChild(ById("list").childNodes[t]); 
	triallist.splice(t,1);
	phiBBseedlist.splice(t,1);
	phiABseedlist.splice(t,1);
	phiBBcontlist.splice(t,1);
	phiABcontlist.splice(t,1);
	ntrial--;
	
	for(t = 0; t < ntrial; t++) ById("num"+triallist[t]).innerHTML = "Group "+(t+1);
	replot();
	calc();
}

function move(e,num,ty)
{	 
	var rect, lim = 0.05;
	switch(ty){
	case 0: rect = ById("seed_trial"+num).getBoundingClientRect(); break;
	case 1: rect = ById("cont_trial"+num).getBoundingClientRect(); break;
	}
	
	mx = e.clientX - rect.left;
	my = e.clientY - rect.top;
	
	t = 0; while(t < ntrial && triallist[t] != "trial"+num) t++;
	if(t == ntrial) alert("cannot find");

	if(mx >= 10 && mx < 60 && my > 35 && my < 95){
		overN = t; overty = ty;
	}
	else{
		overN = -1;
	}
	
	if(butdown == 1){
		phiBB = (mx-px1-(px3-px1)*(py1-my)/(py1-py3))/(px2-px1); if(phiBB < 0 && phiBB > -lim) phiBB = 0; if(phiBB > 1 && phiBB< 1+lim) phiBB = 1;
		phiAB = (py1-my)/(py1-py3); if(phiAB < 0 && phiAB > -lim) phiAB = 0; if(phiAB > 1 && phiAB< 1+lim) phiAB = 1;
		phiAA = 1-phiBB-phiAB;
		
		
		if(phiAA < 0 && phiAA > -lim){
			if(phiBB > phiAB) phiBB += phiAA; else phiAB += phiAA;
			phiAA = 0;
		}

		if(phiAA > 1 && phiAA< 1+lim){
			if(phiBB < phiAB) phiBB += (phiAA-1); else phiAB += (phiAA-1);
			phiAA = 1;
		}
	
		if(phiBB >= 0 && phiBB <= 1 && phiAB >= 0 && phiAB <= 1 && phiAA >= 0 && phiAA <= 1){
			switch(ty){
			case 0:	
				phiBBseedlist[t] = phiBB;
				phiABseedlist[t] = phiAB;
				break;
			case 1:	
				phiBBcontlist[t] = phiBB;
				phiABcontlist[t] = phiAB;
				break;
			}
			drawcan("trial"+num,t,ty);
			calc();
		}
   }
   drawcan("trial"+num,t,ty);	
}

function pr(a){ console.log(a);}

function check_input()
{
	var NI = parseInt(ById("nseeder").value);
	if(NI <= 0){
		alert("# Seeder must be a positive quantity");
		NI = 1;
	}
	ById("nseeder").value = NI;


	var NS = parseInt(ById("ncontact").value);
	if(NS <= 0){
		alert("# Contact must be a positive quantity");
		NS = 1;
	}
	ById("ncontact").value = NS;
		
	var nrep = parseInt(ById("nrep").value);
	if(nrep <= 0){
		alert("# Replicates must be a positive quantity");
		nrep = 1;
	}
	ById("nrep").value = nrep;
	
	var kshape = ById("kshape").value;
	if(kshape <= 0){
		alert("Shape parameter k must be a positive quantity");
		kshape = 1;
	}
	ById("kshape").value = kshape;
}

function calc()
{
	var t, nrep;

	NI = parseInt(ById("nseeder").value);
	
	NS = parseInt(ById("ncontact").value);
		
	nrep = parseInt(ById("nrep").value);
	
	kshape = ById("kshape").value;
	
	dom = parseFloat(ById("dom").value).toFixed(2);
	ById("domval").innerHTML = dom;
	per = parseInt(ById("frac").value);
	ById("percontinf").innerHTML = per;
	f = per/100;
	
	ById("Nexp").innerHTML = (NI+NS)*ntrial*nrep;

	h = NI/(NI+f*NS);

	xiSav = 0; xiS2av = 0; HSav = 0; etaSav = 0; etaS2av = 0;
	for(t = 0; t < ntrial; t++){ xiSav += xiS[t]; xiS2av += xiS[t]*xiS[t]; HSav += HS[t]; etaSav += etaS[t]; etaS2av += etaS[t];}
	xiSav /= ntrial; xiS2av /= ntrial; HSav /= ntrial; etaSav /= ntrial; etaS2av /= ntrial;
	
	av1 = 0; av2 = 0; av3 = 0;
 	av1eta = 0; av2eta = 0; av3eta = 0;
	for(t = 0; t < ntrial; t++){
		av1 += (xiI[t]-xiS[t])*(xiI[t]-xiS[t]);
		av2 += (xiS[t]-xiSav)*(xiI[t]-xiS[t]);
		av3 += xiI[t]-xiS[t];
		av1eta += (etaI[t]-etaS[t])*(etaI[t]-etaS[t]);
		av2eta += (etaS[t]-etaSav)*(etaI[t]-etaS[t]);
		av3eta += etaI[t]-etaS[t];
	}
	av1 /= ntrial; av2 /= ntrial; av3 /= ntrial;
	av1eta /= ntrial; av2eta /= ntrial; av3eta /= ntrial;

	Hglob = 0; etaglob = 0; xiglob = 0;
	for(t = 0; t < ntrial; t++){
		Hglob += HI[t]*NI+HS[t]*f*NS;
		etaglob += etaI[t]*NI+etaS[t]*f*NS;
		xiglob += xiI[t]*NI+xiS[t]*f*NS;
	}
	Hglob /= ntrial*(NI+f*NS); etaglob /= ntrial*(NI+f*NS); xiglob /= ntrial*(NI+f*NS);

	X = (1-h)*av1 - (NI/(f*NS))*Math.log(h)*Math.log(h)*av3*av3;
	Xeta = (1-h)*av1eta - (NI/(f*NS))*Math.log(h)*Math.log(h)*av3eta*av3eta;
	W = Math.log(h)*av2;
	Weta = Math.log(h)*av2eta;
 
  MSS = nrep*ntrial*f*NS*(HSav-xiSav*xiSav);
	MSI = nrep*ntrial*(f*NS*(xiS2av - xiSav*xiSav) - NI*W);
	MII = nrep*ntrial*(f*NS*(xiS2av - xiSav*xiSav) - NI*(2*W-X));
	num = (MSS*MII-MSI*MSI)/MII;

	
	if(num <= 0 || MII == 0) ById("sig_g").innerHTML = "?";
	else ById("sig_g").innerHTML = (1/Math.sqrt(num)).toFixed(3);

	num = (MSS*MII-MSI*MSI)/MSS;
    if(num <= 0 || MSS == 0) ById("sig_f").innerHTML = "?";
	else ById("sig_f").innerHTML = (1/Math.sqrt(num)).toFixed(3);

	num = kshape*nrep*ntrial*(NI+f*NS)*(Hglob - xiglob*xiglob);
	if(num <= 0) ById("sig_r").innerHTML = "?";
	else ById("sig_r").innerHTML = (1/Math.sqrt(num)).toFixed(3);

	num = nrep*ntrial*f*NS*(etaSav - etaSav*etaSav);
	if(num <= 0) st = "?";
	else st = (1/Math.sqrt(num)).toFixed(3);
 	ById("dom_g").innerHTML = st+ "/|<i>a<sub>g</sub></i>|";

	num = nrep*ntrial*f*NS*(etaS2av - etaSav*etaSav) + nrep*ntrial*NI*(-2*Weta + Xeta);
	if(num <= 0) st = "?";
	else st = (1/Math.sqrt(num)).toFixed(3);
 	ById("dom_f").innerHTML = st+ "/|<i>a<sub>f</sub></i>|";

	num = kshape*nrep*ntrial*(NI+f*NS)*(etaglob- etaglob*etaglob);
	if(num <= 0) st = "?";
	else st = (1/Math.sqrt(num)).toFixed(3);
 	ById("dom_r").innerHTML = st+ "/|<i>a<sub>r</sub></i>|";
	
	replot();
}

function drawcan(nam,t,ty)
{
	var typest = "seed"; if(ty == 1) typest = "cont";
	
	NI = parseInt(ById("nseeder").value);
	NS = parseInt(ById("ncontact").value);


	if(model == "VAC"){
		
		var NAA, NBB, NAB = 0;
		var dom = 0;
	
		switch(ty){
		case 0: 
			ById("vac_slider_seed_"+nam).value = phiBBseedlist[t];
	
			NBB = Math.round(phiBBseedlist[t]*NI);
			NAA = NI - NBB;
			
			xiI[t] = (NBB - NAA)/NI;
			HI[t] = 1;
			etaI[t] = 0;
			break;
			
		case 1: 
			ById("vac_slider_cont_"+nam).value = phiBBcontlist[t];
			
			NBB = Math.round(phiBBcontlist[t]*NS);
			NAA = NS - NBB;
			
			xiS[t] = (NBB - NAA)/NS;
			HS[t] = 1;
			etaS[t] = 0;
			break;
		}
		
		var st = "<div style='float:left; font:italic 16px Arial'><i>N<sub>vac</sub></i>= "+NBB+"</div>";
		
		st += "<div style='float:right; font:italic 16px Arial''><i>N<sub>unvac</sub></i>="+NAA+"</div>";
		
		ById(typest+"_"+nam).innerHTML = st;
		return;
	}
	
	var ctx, N;

	dom = parseFloat(ById("dom").value).toFixed(2);
	
	switch(ty){
	case 0:
		ctx = ById("seed_"+nam).getContext("2d");

		NBBf = phiBBseedlist[t]*NI;
		NABf = phiABseedlist[t]*NI;
		NAAf = (1-phiBBseedlist[t]-phiABseedlist[t])*NI;

		min = 100000;
		for(NAAc = Math.round(NAAf)-1; NAAc <= Math.round(NAAf)+1; NAAc++){
			for(NABc = Math.round(NABf)-1; NABc <= Math.round(NABf)+1; NABc++){
				NBBc = NI - NABc - NAAc;
				if(NAAc >= 0 && NABc >= 0 && NBBc >= 0){
					d = (NAAf-NAAc)*(NAAf-NAAc) + (NABf-NABc)*(NABf-NABc) + (NBBf-NBBc)*(NBBf-NBBc);
					if(d < min){ min = d; NAA = NAAc; NAB = NABc; NBB = NBBc;}
				}
			}
		}
		
		N = NI;
		break;
	case 1:
		ctx = ById("cont_"+nam).getContext("2d");

		NBBf = phiBBcontlist[t]*NS;
		NABf = phiABcontlist[t]*NS;
		NAAf = (1-phiBBcontlist[t]-phiABcontlist[t])*NS;

		min = 100000;
		for(NAAc = Math.round(NAAf)-1; NAAc <= Math.round(NAAf)+1; NAAc++){
			for(NABc = Math.round(NABf)-1; NABc <= Math.round(NABf)+1; NABc++){
				NBBc = NS - NABc - NAAc;
				if(NAAc >= 0 && NABc >= 0 && NBBc >= 0){
					d = (NAAf-NAAc)*(NAAf-NAAc) + (NABf-NABc)*(NABf-NABc) + (NBBf-NBBc)*(NBBf-NBBc);
					if(d < min){ min = d; NAA = NAAc; NAB = NABc; NBB = NBBc;}
				}
			}
		}
		
		N = NS;
		break;
	}
	
	ctx.clearRect(0,0,dx,dy);

	ctx.strokeStyle = "black";
	ctx.beginPath();
	ctx.moveTo(px1,py1);
	ctx.lineTo(px2,py2);
	ctx.lineTo(px3,py3);
	ctx.lineTo(px1,py1);
	ctx.fillStyle = "white";
	ctx.fill(); 
	ctx.stroke(); 
		
	NN = N; while(NN > 30) NN = Math.round(NN/2);
	
	ctx.strokeStyle = "#aaaaaa";
	for(i = 1; i < NN; i++){
		ctx.beginPath();
		ctx.moveTo(px1+(px2-px1)*i/NN,py1+(py2-py1)*i/NN);
		ctx.lineTo(px1+(px3-px1)*i/NN,py1+(py3-py1)*i/NN);
		ctx.stroke(); 
	}
		
	for(i = 1; i < NN; i++){
		ctx.beginPath();
		ctx.moveTo(px2+(px3-px2)*i/NN,py2+(py3-py2)*i/NN);
		ctx.lineTo(px2+(px1-px2)*i/NN,py2+(py1-py2)*i/NN);
		ctx.stroke(); 
	}
		
	for(i = 1; i < NN; i++){
		ctx.beginPath();
		ctx.moveTo(px3+(px1-px3)*i/NN,py3+(py1-py3)*i/NN);
		ctx.lineTo(px3+(px2-px3)*i/NN,py3+(py2-py3)*i/NN);
		ctx.stroke(); 
	}
		
	ctx.beginPath();
	ctx.arc(px1+NBB*(px2-px1)/N + NAB*(px3-px1)/N,py1+NBB*(py2-py1)/N + NAB*(py3-py1)/N,7,0,2*Math.PI);
	ctx.fillStyle = "purple";
	ctx.fill();

	ctx.font = "16px Arial";
	ctx.fillStyle = "red";
	ctx.fillText("AA",0,dy-8);
	ctx.fillText("A",px3-10,py3-4);
	
	ctx.fillStyle = "blue";
	ctx.fillText("BB",dx-21,dy-8); 
	ctx.fillText("B",px3+1,py3-4);
		
	ctx.fillStyle = "black";
	
	ctx.font = "italic 16px Arial";
		
	switch(ty){
	case 0: ctx.fillText("Seeders:",1,15); break;
	case 1: ctx.fillText("Contacts:",1,15); break;
	}
	
	ctx.fillStyle = "black"; ctx.font = "italic 16px Arial";
	if(overN == t) ctx.fillStyle = "grey";

	y = 50;
	ctx.font = "italic 16px Arial"; ctx.fillText("N",10,y);
	ctx.font = "10px Arial"; ctx.fillText("AA",22,y+3);
	ctx.font = "16px Arial"; ctx.fillText("="+NAA,35,y);
		
	y = 70;
	ctx.font = "italic 16px Arial"; ctx.fillText("N",10,y);
	ctx.font = "10px Arial"; ctx.fillText("BB",22,y+3);
	ctx.font = "16px Arial"; ctx.fillText("="+NBB,35,y);
	
	y = 90;
	ctx.font = "italic 16px Arial"; ctx.fillText("N",10,y);
	ctx.font = "10px Arial"; ctx.fillText("AB",22,y+3);
	ctx.font = "16px Arial"; ctx.fillText("="+NAB,35,y);

	ctx.fillStyle = "black";

	switch(ty){
	case 0:
		xi = (NBB + dom*NAB - NAA)/NI; xiI[t] = xi;
		H = (NAA + dom*dom*NAB + NBB)/NI; HI[t] = H;
	 	etaI[t] = NAB/NI;
		break;
	case 1:
		xi = (NBB + dom*NAB - NAA)/NS; xiS[t] = xi;
		H = (NAA + dom*dom*NAB + NBB)/NS; HS[t] = H;
		etaS[t] = NAB/NS;
		break;
	}
	
	y = 50;
	ctx.font = "italic 18px Times"; ctx.fillText("\u03c7",dx-57,y);
	ctx.font = "16px Arial"; ctx.fillText("="+xi.toFixed(2),dx-46,y);
	y = 70;
	ctx.font = "italic 16px Times"; ctx.fillText("H",dx-57,y);
	ctx.font = "16px Arial"; ctx.fillText("="+H.toFixed(2),dx-46,y);
}		

function replot()
{
	if(newtype == "User defined"){
		var w;
		switch(ntrial){
		case 1: w = 280; break;
		case 2: w = 550; break;
		default: w = 800; break;
		}
		ById("wrap").style.width = w+"px";
	}
		
	ById("output_box").style.left = 300 + ById("wrap").clientWidth+"px";
	
	for(t = 0; t < ntrial; t++){ na = triallist[t]; drawcan(na,t,0); drawcan(na,t,1);}	
}

function init()
{
	model = "SNP"; document.forms["formMOD"]["type"].value = "SNP";
	
	document.forms["formSNP"]["type"].value = "Single";
	document.forms["formVAC"]["type"].value = "Single";
 
	ById("frac").value = 100;
	ById("dom").value = 0;
	
	changetype();
}

function changentrial()
{
	if(model == "SNP") ntrialnew = parseInt(ById("ntrialSNP").value);
	else ntrialnew = parseInt(ById("ntrialVAC").value);
 
	if(isNaN(ntrialnew)){ alert("Not a number!");}
	else{
		if(ntrialnew < 1 || ntrialnew > 100) alert("Not in range!");
		else{
			if(ntrialnew >  ntrial){
				while(ntrialnew > ntrial) addtrial();
			}
			else{
				while(ntrialnew < ntrial) removetrial(ntrial-1);
			}
			if(model == "SNP") ById("ntrialSNP").value = ntrial;
			else ById("ntrialVAC").value = ntrial;
			
			replot();
			calc();
		}
	}
}

function changemodel()
{
	model = document.forms["formMOD"]["type"].value;

	if(model == modelold) return;
	
	modelold = model;

	if(model == "SNP"){
		document.forms["formSNP"].style.display = "inline";
		document.forms["formVAC"].style.display = "none";
		ById("dom_g_show").style.display = "";
		ById("dom_f_show").style.display = "";
		ById("dom_r_show").style.display = "";
		ById("dom_show").style.display = "";
		ById('input_box').style.height = "415px"; 
		ById('output_box').style.height = "230px";
	}
	else{
		document.forms["formSNP"].style.display = "none";
		document.forms["formVAC"].style.display = "inline";
		ById("dom_g_show").style.display = "none";
		ById("dom_f_show").style.display = "none";
		ById("dom_r_show").style.display = "none";
		ById("dom_show").style.display = "none";
		ById('input_box').style.height = "335px"; 
		ById('output_box').style.height = "143px";
	}
	
	oldtype = null;
	changetype();
}

function changetype()
{
	var t;

	if(model == "SNP") newtype = document.forms["formSNP"]["type"].value;
	else newtype = document.forms["formVAC"]["type"].value;
	
	if(newtype == oldtype) return;
	oldtype = newtype;
	
	if(newtype == "User defined"){
		ById("ntrSNP").style.visibility = "visible";		
		ById("ntrVAC").style.visibility = "visible";		
	}
	else{
		ById("ntrSNP").style.visibility = "hidden";
		ById("ntrVAC").style.visibility = "hidden";
	}
	
	ById("list").innerHTML = "";
	ntrial = 0;	
	triallist.length=0; phiBBseedlist.length=0; phiABseedlist.length=0; phiBBcontlist.length=0; phiABcontlist.length=0;
		
	switch(newtype){
	case "Single":
		ById("wrap").style.width = "280px";
	
		ById("nseeder").value = 28;
		ById("ncontact").value = 172;
		ById("nrep").value = 1;
	
		addtrial();
		phiBBseedlist[0]=0.0; phiABseedlist[0]=0; phiBBcontlist[0]=0.95; phiABcontlist[0]=0;
		break;

	case "Pure":
		ById("wrap").style.width = "550px";
	
		ById("nseeder").value = 12;
		ById("ncontact").value = 13;
		ById("nrep").value = 2;
		for(t = 0; t < 4; t++) addtrial();

		phiBBseedlist[0]=0; phiABseedlist[0]=0; phiBBcontlist[0]=0; phiABcontlist[0]=0;
		phiBBseedlist[1]=0; phiABseedlist[1]=0; phiBBcontlist[1]=1; phiABcontlist[1]=0;
		phiBBseedlist[2]=1; phiABseedlist[2]=0; phiBBcontlist[2]=0; phiABcontlist[2]=0;
		phiBBseedlist[3]=1; phiABseedlist[3]=0; phiBBcontlist[3]=1; phiABcontlist[3]=0;
		break;
		
	case "Puredom":
		ById("wrap").style.width = "800px";
	
		ById("nseeder").value = 11;
		ById("ncontact").value = 11;
		ById("nrep").value = 1;
		for(t = 0; t < 9; t++) addtrial();

		phiBBseedlist[0]=0; phiABseedlist[0]=0; phiBBcontlist[0]=0; phiABcontlist[0]=0;
		phiBBseedlist[1]=0; phiABseedlist[1]=1; phiBBcontlist[1]=0; phiABcontlist[1]=0;
		phiBBseedlist[2]=1; phiABseedlist[2]=0; phiBBcontlist[2]=0; phiABcontlist[2]=0;
		phiBBseedlist[3]=0; phiABseedlist[3]=0; phiBBcontlist[3]=0; phiABcontlist[3]=1;
		phiBBseedlist[4]=0; phiABseedlist[4]=1; phiBBcontlist[4]=0; phiABcontlist[4]=1;
		phiBBseedlist[5]=1; phiABseedlist[5]=0; phiBBcontlist[5]=0; phiABcontlist[5]=1;
		phiBBseedlist[6]=0; phiABseedlist[6]=0; phiBBcontlist[6]=1; phiABcontlist[6]=0;
		phiBBseedlist[7]=0; phiABseedlist[7]=1; phiBBcontlist[7]=1; phiABcontlist[7]=0;
		phiBBseedlist[8]=1; phiABseedlist[8]=0; phiBBcontlist[8]=1; phiABcontlist[8]=0;
		break;
	
	case "Mixed":
		ById("wrap").style.width = "550px";
	
		ById("nseeder").value = 2;
		ById("ncontact").value = 23;
		ById("nrep").value = 4;
		for(t = 0; t < 2; t++) addtrial();

		phiBBseedlist[0]=0; phiABseedlist[0]=0; phiBBcontlist[0]=0.146; phiABcontlist[0]=0;
		phiBBseedlist[1]=1; phiABseedlist[1]=0; phiBBcontlist[1]=0.854; phiABcontlist[1]=0;
		break;
		
	case "Mixeddom":
		ById("wrap").style.width = "800px";
	
		ById("nseeder").value = 2;
		ById("ncontact").value = 15;
		ById("nrep").value = 4;
		for(t = 0; t < 3; t++) addtrial();

		phiBBseedlist[0]=0; phiABseedlist[0]=0; phiBBcontlist[0]=0.11; phiABcontlist[0]=0.11;
		phiBBseedlist[1]=0; phiABseedlist[1]=1; phiBBcontlist[1]=0.11; phiABcontlist[1]=0.78;
		phiBBseedlist[2]=1; phiABseedlist[2]=0; phiBBcontlist[2]=0.78; phiABcontlist[2]=0.11;
		break;

	case "User defined":
		ById("wrap").style.width = "800px";
		
		ById("nseeder").value = 2;
		ById("ncontact").value = 8;
		ById("nrep").value = 20;
	
		addtrial();
		phiBBseedlist[0]=0.0; phiABseedlist[0]=0; phiBBcontlist[0]=0; phiABcontlist[0]=0;
		break;
	}
	
		
	if(model == "SNP") ById("ntrialSNP").value = ntrial;
	else ById("ntrialVAC").value = ntrial;
	
	replot();
	calc();
}

function inputon()
{
	t = overN;
	nam = triallist[t];
	NI = parseInt(ById("nseeder").value);
	NS = parseInt(ById("ncontact").value);
	
	switch(overty){
	case 0:
		NBB = Math.round(phiBBseedlist[t]*NI);
		NAB = Math.round(phiABseedlist[t]*NI);
		NAA = NI-NBB-NAB;
		N = NI;
		ById("overcat").innerHTML = "seeders";
		break;
	case 1:
		NBB = Math.round(phiBBcontlist[t]*NS);
		NAB = Math.round(phiABcontlist[t]*NS);
		NAA = NS-NBB-NAB;
		N = NS;
		ById("overcat").innerHTML = "contacts";
		break;
	}
	ById("NAA").value = NAA;
	ById("NBB").value = NBB;
	ById("NAB").innerHTML = NAB;
		
	ById("trialnum").innerHTML = (overN+1);

	ById("shad").style.visibility= "visible";
	ById("inp").style.visibility= "visible";
}

function inputoff(i)
{
	t = overN;
	nam = triallist[t];
	NI = parseInt(ById("nseeder").value);
	NS = parseInt(ById("ncontact").value);
	
	
	if(i == 1){
		NAA = parseInt(ById("NAA").value);
		NBB = parseInt(ById("NBB").value);
		NAB = parseInt(ById("NAB").innerHTML);
		
		switch(overty){
		case 0: NAB = NI-NAA-NBB; break;
		case 1: NAB = NS-NAA-NBB; break;
		}
		
		if(isNaN(NAA) || isNaN(NBB) || isNaN(NAB)){ alert("Entries not valid"); return;}
		if(NAA < 0 || NBB < 0 || NAB < 0){ alert("Entries must be positive"); return;}
		
		switch(overty){
		case 0: phiBBseedlist[t] = NBB/NI;	 phiABseedlist[t] = NAB/NI;	break;
		case 1: phiBBcontlist[t] = NBB/NS;	 phiABcontlist[t] = NAB/NS;	break;
		}
	}
	overN=-1; ById('shad').style.visibility= 'hidden'; ById('inp').style.visibility= 'hidden';
	replot();
	calc();
}

function  helpon()
{
	ById('shad').style.visibility= 'visible'; ById('help').style.visibility= 'visible';
}

function  helpoff()
{
	ById('shad').style.visibility= 'hidden'; ById('help').style.visibility= 'hidden';
}


function  paramdefon()
{
	ById('shad').style.visibility= 'visible'; ById('paramdef').style.visibility= 'visible';
}

function  paramdefoff()
{
	ById('shad').style.visibility= 'hidden'; ById('paramdef').style.visibility= 'hidden';
}



function recal()
{
	NI = parseInt(ById("nseeder").value);
	NS = parseInt(ById("ncontact").value);
	
	NAA = parseInt(ById("NAA").value);
	NBB = parseInt(ById("NBB").value);
	NAB = parseInt(ById("NAB").innerHTML);
		
	switch(overty){
	case 0: NAB = NI-NAA-NBB; break;
	case 1: NAB = NS-NAA-NBB; break;
	}
	ById("NAB").innerHTML = NAB;
}

function ById(a){ return  document.getElementById(a);}
</script>
</head>

<body onload="init();">
<img src="SIREPClogo.png" height=90 />
<br/>
<div style="width:850px; margin-top:10px; margin-bottom:0px;">
This calculator tool estimates precisions for SNP or vaccination-based effects that alter individual susceptibility, infectivity and/or recoverability using infection and recovery time data from a disease transmission experiment.
</div>
<div style="width:850px; margin-top:10px; margin-bottom:0px;">
For further details please click <a href="javascript:helpon();">here</a> and for parameter definitions  <a href="javascript:paramdefon();">here</a>.
</div>
<br/> 

<div style='float:left; width:300px;'>
 
 
 
<div id='input_box' class='boxgreen' style='position:absolute; left:10px; top:185px; width:270px; height:415px; margin:8px; font-size:17px;'>
<center><b style='font-size:20px'>INPUTS</b></center>
 <b style='font-size:12px; margin:0px;padding:0px'>EFFECTS:</b> <br>
	<form name="formMOD" onsubmit="return false" style="margin-top:1px; margin-bottom:8px;" onchange="changemodel();">
	<input type="radio" name="type" value="SNP"> SNP <br>
	<input type="radio" name="type" value="VAC"> Vaccination <br>
	</form>
	
 <b style='font-size:12px; margin:0px; padding:0px'>DESIGN:</b> <br>
 
<form name="formSNP"  onsubmit="return false" style="margin-top:0px; margin-bottom:8px;" onchange="changetype();">
 <input type="radio" name="type" value="Single"> Single group (w/o dominance) <br>
 <input type="radio" name="type" value="Pure"> Pure design (w/o dominance)<br>
 <input type="radio" name="type" value="Puredom"> Pure design (with dominance)<br>
 <input type="radio" name="type" value="Mixed" > Mixed design (w/o dominance)<br>
 <input type="radio" name="type" value="Mixeddom" > Mixed design (with dominance) <br>
 <input type="radio" name="type" value="User defined"> User defined <span id="ntrSNP" style="visibility:hidden"> - Groups: <input  type="number" style="width:40px; margin-right:10px" id="ntrialSNP" value="1" step="1" min="1" max="100" onchange="check_input(); changentrial()"></span>
</form>

<form name="formVAC"  onsubmit="return false" style="display:none; margin-top:8px; margin-bottom:0px;" onchange="changetype();">
 <input type="radio" name="type" value="Single"> Single group<br>
 <input type="radio" name="type" value="Pure"> Pure design<br>
 <input type="radio" name="type" value="Mixed"> Mixed design<br>
 <input type="radio" name="type" value="User defined"> User defined <span id="ntrVAC" style="visibility:hidden"> - Groups: <input  type="number" style="width:40px; margin-right:3px" id="ntrialVAC" value="1" step="1" min="1" max="100" onchange="changentrial()"></span>
</form>

 <b style='font-size:12px; margin:0px;padding:0px'>DETAILS:</b> <br>
<table>
<tr>
<td style="vertical-align:top"># Seeders per group:</td><td style="vertical-align:top"><input type="number" style="width:50px; margin-right:30px;" id="nseeder" value=1 step="1" min="1" max="1000" onchange="check_input(); replot(); calc();"></td>
</tr>
<tr>
<td style="vertical-align:top"># Contacts per group:</td><td style="vertical-align:top"> <input type="number" style="width:50px; margin-right:30px" id="ncontact" value = 10 step="1" min="1" max="1000" onchange="check_input(); replot(); calc();"></td>
</tr>
<tr>
<td style="vertical-align:top"># Replicates:</td><td style="vertical-align:top"> <input type="number" style="width:50px; margin-right:30px" id="nrep" value = 1 step="1" min="1" max="100" onchange="check_input(); calc();"></td>
</tr>
<tr>
<td style="vertical-align:top">Shape parameter <i>k</i>:</td><td style="vertical-align:top"> <input type="number" style="width:50px; margin-right:30px" id="kshape" value = 5 step="1" min="1" onchange="check_input(); calc();"></td>
</tr>
<tr>
<td style="vertical-align:top">% Infected <i>&straightphi;</i>:<span style='font-size:16px' id='percontinf'></span> </td><td style="vertical-align:top"> &nbsp;<input id='frac' type='range'  style="width:100px; height:20px; margin:0px; padding:0px; margin-left:0px;" min='1' max = '100' step = '1' onmouseup='calc()' oninput='calc()' value = 100></td>
</tr>
<tr id="dom_show">
<td style="vertical-align:top">Dominance: <span style='font-size:16px' id='domval'></span>   </td><td style="vertical-align:top">  <span style='color:red;'><b>A</b></span><input id='dom' type='range' style="width:70px; height:20px; margin:0px; padding:0px; margin-left:5px;  margin-right:5px;" min='-1' max = '1' step = 0.1 onmouseup='replot(); calc();' oninput='replot(); calc()' value = 0><span style='color:blue;'><b>B</b></span></td>
</tr>
</table>
</div>

<div id="output_box" class='boxred' style='position:absolute; left:610px; top:185px; width:270px; height:230px; margin:8px; font-size:17px;'>
<center><b style='font-size:20px, font-family:Arial'>OUTPUTS</b></center>
<table>
<tr><td width = 150><i style="font-family:Arial">N<sub>total</sub></i> = <span id="Nexp"></span></td><td width = 110>Individuals</td></tr>
<tr><td style="vertical-align:top">SD in <i>a<sub>g</sub></i> = <span id="sig_g"></span></td><td style="vertical-align:top">(Susceptibility)</td></tr>
<tr><td style="vertical-align:top">SD in <i>a<sub>f</sub></i>  = <span id="sig_f"></span></td><td style="vertical-align:top">(Infectivity)</td></tr>
<tr><td style="vertical-align:top">SD in <i>a<sub>r</sub></i>  = <span id="sig_r"></span></td><td style="vertical-align:top">(Recovery)</td></tr>
<tr id="dom_g_show"><td style="vertical-align:top">SD in &Delta;<i><sub>g</sub></i> = <span id="dom_g"></span></td><td style="vertical-align:top">(Dom. in Susc.)</td></tr>
<tr id="dom_f_show"><td style="vertical-align:top">SD in &Delta;<i><sub>f</sub></i>  = <span id="dom_f"></span></td><td style="vertical-align:top">(Dom. in Inf.)</td></tr>
<tr id="dom_r_show"><td style="vertical-align:top">SD in &Delta;<i><sub>r</sub></i>  = <span id="dom_r"></span></td><td style="vertical-align:top">(Dom. in Rec.)</td></tr>
</table>

</div>
</div>

<div id="wrap" style="position:absolute; left:310px; top:185px; height:50px; ">
<span id="list"></span>
</div>

<div class="shad" id="shad" style="position:fixed; background:#000044; opacity:0.5; visibility:hidden"></div>
<div class ="inp" id="inp" style=" background:#ffffff; visibility:hidden; padding:20px">
<center> <b>Number of <span id="overcat"></span> in group <span id="trialnum"></span>:</b><br/><br/>
<table>
<tr><td style="vertical-align:top">N<sub>AA</sub></td><td style="vertical-align:top">=</td><td style="vertical-align:top"><input onchange="recal()" id="NAA" type="number" style="width:150px;"/><td style="vertical-align:top"></tr>
<tr><td style="vertical-align:top">N<sub>BB</sub></td><td style="vertical-align:top">=</td><td style="vertical-align:top"><input onchange="recal()" id="NBB" type="number" style="width:150px;"/><td style="vertical-align:top"></tr>
<tr><td style="vertical-align:top">N<sub>AB</sub></td><td style="vertical-align:top">=</td><td style="vertical-align:top"><span  id="NAB" type="number" style="width:150px;"></span><td style="vertical-align:top"></tr>
</table>
<br/>
 <input id="number" type="button" value="OK" onclick="inputoff(1);"" style="width:100px; float:right;"/>
 <input id="number" type="button" value="Cancel" onclick="inputoff(0);" style="width:100px; float:left;"/><br/>
</center> 
</div>

<div class ="help" id="help" style=" background:#ffffff; visibility:hidden; padding:20px;  ">
<i style="font-size:25px;">Help</i>
<div class="close" style='color:red; width:30px;'><a style='color:red; text-decoration:none;' href='javascript: helpoff();'>✖</a></div>

<div class="inside" style="overflow:scroll; height:85%; margin-top:10px">

<h2>What are SNP and vaccination effects? </h2>
<p> Vaccinations are known to be an effective way to reduce the probability of becoming infected, however they will not protect every individual. Furthermore, once infected a vaccinated individual may be less likely to transmit the disease and/or recover faster. Understanding these mechanisms is crucial to estimating the effectiveness of a vaccination campaign. We define vaccination "effects" as the average difference in three key epidemiological traits between the vaccinated and unvaccinated groups: susceptibility, infectivity and recoverability.
</p>

<p>
 Single nucleotide polymorphisms (SNP) capture individual variability at a particular genetic nucleotide. Some SNPs have been discovered that have large "effects" on the three host epidemiological traits. Targeting these SNPs in selective breeding programmes offers a potential avenue for effective disease control. Here it is assumed that the SNP under consideration takes two possible alleles (<i>A</i> and <i>B</i>), and that individuals are diploid (i.e. they have genotypes <i>AA</i>, <i>AB</i> or <i>BB</i>).
</p>

<p>
 The property of "dominance" captures the degree to which one allele (the dominant one) masks the phenotypic contribution of the other (recessive). For example, if <i>A</i> is dominant then individuals with genotype <i>AA</i> and <i>AB</i> will behave in a similar way, but those with genotype <i>BB</i> will behave differently.
 </p>

<h2> What is the purpose of <i>SIRE-PC</i>?</h2>
<p>
We begin by assuming that a disease transmission experiment is going to be undertaken to establish how a particular SNP (or vaccination status) affects the three host traits: susceptibility, infectivity and recoverability. 
</p>
<p>Existing computational tools (such as <a href="https://theiteam.github.io/SIRE.html">SIRE</a>) make it possible to estimate variation in these traits using the infection and recovery times of individuals in the experiment (or alternatively just recovery times or disease diagnostic test results, although these lead to an additional loss in statistical power not considered here).
</p>

<p>
A key question to ask is how should the experiment be best designed in order to maximise the precision with which SNP/vaccination effects can be made? It is this question that <i>SIRE-PC</i> aims to answer.
</p>

 
 <h2> How do I use <i>SIRE-PC</i>?</h2>
 
 <p>
 The instructions below follow the case of SNP effects, and a description of how these are modified for vaccination effects is provided later.
</p> 
 
<p> The tool works in the following way:</p>

<h3>Inputs</h3>
<p>
 The first option to select is the type of effects being investigated (either SNP or vaccination).
 </p>
 
 <p>
 The next option selects the experimental design to be carried out. These include the five potential designs outlined in the paper: "Single group (without dominance)", "Pure design (without dominance)", "Pure design (with dominance)", "Mixed design (without dominance)" and "Mixed design (with dominance)".
 </p>
 
<p>
 Selecting "User defined" allows the user to set an arbitrary design. Here the number of groups can be altered by entering the relevant number in the box (also groups can be removed by clicking on the "x" symbols in the corner).
</p>

 <p>The number of seeders and contacts refers to the number of individuals initially infected and susceptible in each contact group. The genetic compositions of these subpopulations is shown by the panels on the right (one for each contact group). Here <i>N<sub>AA</sub></i>, <i>N<sub>BB</sub></i> and <i>N<sub>AB</sub></i> refer to the number of individuals in each of the three genotypes. These genetic compositions may be changed by either clicking on the numbers themselves, or by dragging the purple circles within the triangle plots.  
 </p>
 
 <p>
Often the same basic design is repeated multiple times as a way of increasing statistical power. This duplication is represented by the number of "replicates".
 </p>
 
 <p>
 The shape parameter <i>k</i> governs the gamma distributed recovery profile (note this quantity only affects estimates for recovery rate parameters).
 </p>
 <p>
 The % infected slider sets <i>&straightphi;</i>, the expected fraction of contacts that become infected during the course of the experiment. This may be significantly less than 100% if the experiment is terminated early or the basic reproductive ratio <i>R<sub>0</sub></i> is low.
 </p>
 <p> 
 In some circumstances allele <i>A</i> may dominate over <i>B</i>, or <i>vice-versa</i>. This affect can be accounted for by adjusting the relevant dominance slider.
 </p>
 
<h3>Outputs</h3>
<p>
The first number shows the total number of individuals for the experiment. 
</p>
<p>
Model parameters <i>a<sub>g</sub></i>, <i>a<sub>f</sub></i>, and <i>a<sub>r</sub></i> represent the relative differences in susceptibility, infectivity and recovery rate for individuals with an <i>A</i> compared to a <i>B</i> allele at the SNP under investigation. For example, <i>a<sub>g</sub></i>=0.1 represents the case in which individuals with genotype <i>AA</i> are approximately 20% more susceptible to disease than those with <i>BB</i> (see paper for a more precise definition).
</p>
<p>
Due to the stochastic nature of data obtained from disease transmission experiments (<i>i.e.</i> infection and recovery times), precise estimates for <i>a<sub>g</sub></i>, <i>a<sub>f</sub></i>, and <i>a<sub>r</sub></i> are not possible. What this tool shows is the expected standard deviation in the posterior distribution of these quantities, which provide estimates for their precision. Small numbers represent a higher degree of precision, and so the experimental design should be chosen to minimise these standard deviations as much as possible.
</p>

<p>
Parameters &Delta;<i><sub>g</sub></i>, &Delta;<i><sub>f</sub></i>, and &Delta;<i><sub>r</sub></i> represent the scaled dominance factors of allele <i>A</i> over <i>B</i>. Note, the standard deviations in these quantities are each divided by the corresponding effect size (because if the SNP effect size is small, it becomes harder to establish dominance).
</p>

<h3>Vaccination effects</h3>
<p>
The key difference between studying vaccination effects and SNP effects is that instead of three genotypes (<i>AA</i>, <i>AB</i> and <i>BB</i>) we have two classifications: unvaccinated and vaccinated. Effectively this means we can associated unvaccinated individuals to be represented by <i>AA</i>, vaccinated by <i>BB</i> and set the number of <i>AB</i> individuals to zero. 
This also means that dominance does not need to be considered, and so experimental designs aimed at estimating this are omitted (along with the input slider giving the expected dominance of <i>A</i> over <i>B</i>). Furthermore, instead of triangular plots we now simply use sliders to control the vaccination status composition for the seeder and contact populations within each group.
</p>

<h3>General application</h3>
<p>
Two important points need to be made when considering the general application of the power calculations presented here:
</p>
<p>
<b>•</b> It is assumed that individuals are randomly distributed regarding the effects of other SNPs on the 3 parameters. In particular, it is assumed that related individuals are randomly distributed across groups.
</p>
<p>
<b>•</b> The estimates provided by this tool represent lower bounds on the actual standard deviations of model parameters. As shown in the "Realistic model and data scenarios" section of the paper, residual contributions, group and fixed effects and incomplete data will all act to increase these standard deviations, and so reduce the statistical power with which SNP and vaccination-based associations can be made.
 </p>
 </div>
 
 
  <input class="br" id="number" type="button" value="OK" onclick="helpoff();"" style="width:100px;"/>

</div>

<div class ="paramdef" id="paramdef" style=" background:#ffffff; visibility:hidden; padding:20px;  ">
<i style="font-size:25px;">Definitions</i>
<div class="close" style='color:red; width:30px;'><a style='color:red; text-decoration:none;' href='javascript: paramdefoff();'>✖</a></div>

<div class="inside" style="overflow:scroll; height:85%; margin-top:10px">

<table style="vertical-align:top">
<tr><td style="vertical-align:top"><b>SNP</b></td><td width=30></td><td style="vertical-align:top">Single nucleotide polymorphism</td></tr>
<tr style="height:5px"></tr>
<tr><td style="vertical-align:top"><b>Seeder</b></td><td width=30></td><td style="vertical-align:top">Initially infected individual</td></tr>
<tr style="height:5px"></tr>
<tr><td style="vertical-align:top"><b>Contact</b></td><td width=30></td><td style="vertical-align:top">Initially susceptible individual</td></tr>
<tr style="height:5px"></tr>
<tr><td style="vertical-align:top"><b>Dominance </b></td><td width=30></td><td style="vertical-align:top">The degree to which one allele (dominant) masks the phenotypic contribution of the other (recessive)</td></tr>
<tr style="height:5px"></tr>
<tr><td style="vertical-align:top"><b>Replicates</b></td><td width=30></td><td style="vertical-align:top">Number of times an experimental design is copied (to increase statistical power)</td></tr>
<tr style="height:5px"></tr>

<tr><td style="vertical-align:top"><b style="font-family:arial"><i>N<sub>AA</sub></i></b></td><td width=30></td><td style="vertical-align:top">Number of individuals with SNP genotype <i>AA</i></td></tr>
<tr style="height:5px"></tr>
<tr><td style="vertical-align:top"><b style="font-family:arial"><i>N<sub>AB</sub></i></b></td><td width=30></td><td style="vertical-align:top">Number of individuals with SNP genotype <i>AB</i></td></tr>
<tr style="height:5px"></tr>
<tr><td style="vertical-align:top"><b style="font-family:arial"><i>N<sub>BB</sub></i></b></td><td width=30></td><td style="vertical-align:top">Number of individuals with SNP genotype <i>BB</i></td></tr>
<tr style="height:5px"></tr>
<tr><td style="vertical-align:top"><b style="font-family:arial"><i>N<sub>vac</sub></i></b></td><td width=30></td><td style="vertical-align:top">Number of vaccinated individuals</td></tr>
<tr style="height:5px"></tr>
<tr><td style="vertical-align:top"><b style="font-family:arial"><i>N<sub>unvac</sub></i></b></td><td width=30></td><td style="vertical-align:top">Number of unvaccinated individuals</td></tr>
<tr style="height:5px"></tr>
<tr><td style="vertical-align:top"><b style="font-family:arial"><i>N<sub>total</sub></i></b></td><td width=30></td><td style="vertical-align:top">The total number of individuals in the disease transmission experiment</td></tr>

<tr style="height:5px"></tr>

<tr><td style="vertical-align:top"><b style="font-family:times"><i>k</i></b></td><td width=30></td><td style="vertical-align:top">Shape parameter that determines distribution in infectious duration (assumed gamma distributed)</td></tr>
<tr style="height:5px"></tr>
<tr><td style="vertical-align:top"><b style="font-family:times"><i>ϕ</i></b></td><td width=30></td><td style="vertical-align:top">Fraction of contacts assumed to be infected</td></tr>


<tr style="height:5px"></tr>
<tr><td style="vertical-align:top"><b style="font-family:times"><i>χ</i></b></td><td width=30></td><td style="vertical-align:top">Homozygote balance - the proportion of <i>AA</i> individuals minus the proportion of <i>BB</i> individuals</td></tr>

<tr style="height:5px"></tr>
<tr><td style="vertical-align:top"><b style="font-family:times"><i>H</i></b></td><td width=30></td><td style="vertical-align:top">Homozygosity - the proportion of <i>AA</i> individuals plus the proportion of <i>BB</i> individuals</td></tr>

<tr style="height:5px"></tr>
<tr><td style="vertical-align:top"><b style="font-family:times"><i>a<sub>g</sub></i></b></td><td width=30></td><td style="vertical-align:top">SNP effect for susceptibility - half the fractional change comparing the <i>AA</i> and <i>BB</i> genotypes (or unvaccinated/vaccinated)</td></tr>

<tr><td style="vertical-align:top"><b style="font-family:times"><i>a<sub>f</sub></i></b></td><td width=30></td><td style="vertical-align:top">SNP effect for infectivity - half the fractional change comparing the <i>AA</i> and <i>BB</i> genotypes (or unvaccinated/vaccinated)</td></tr>

<tr><td style="vertical-align:top"><b style="font-family:times"><i>a<sub>r</sub></i></b></td><td width=30></td><td style="vertical-align:top">SNP effect for recoverability - half the fractional change comparing the <i>AA</i> and <i>BB</i> genotypes (or unvaccinated/vaccinated)</td></tr>

<tr><td style="vertical-align:top"><b style="font-family:times">SD in <i>a<sub>g</sub></i></b></td><td width=30></td><td style="vertical-align:top">Expected standard deviation in the posterior distribution for <i>a<sub>g</sub></i> - see below for interpretation</td></tr>

<tr><td style="vertical-align:top"><b style="font-family:times">SD in <i>a<sub>f</sub></i></b></td><td width=30></td><td style="vertical-align:top">Expected standard deviation in the posterior distribution for <i>a<sub>f</sub></i> - see below for interpretation</td></tr>

<tr><td style="vertical-align:top"><b style="font-family:times">SD in <i>a<sub>r</sub></i></b></td><td width=30></td><td style="vertical-align:top">Expected standard deviation in the posterior distribution for <i>a<sub>r</sub></i>  - see below for interpretation</td></tr>


<tr style="height:5px"></tr>
<tr><td style="vertical-align:top"><b style="font-family:times">Δ<sub>g</sub></b></td><td width=30></td><td style="vertical-align:top">Susceptibility effect scaled dominance factor (1 = <i>A</i> is completely dominant over <i>B</i>, -1 = <i>B</i> is dominant over <i>A</i>, 0 = no dominance)</td></tr>

<tr style="height:5px"></tr>
<tr><td style="vertical-align:top"><b style="font-family:times">Δ<sub>f</sub></b></td><td width=30></td><td style="vertical-align:top">Infectivity effect scaled dominance factor (1 = <i>A</i> is completely dominant over <i>B</i>, -1 = <i>B</i> is dominant over <i>A</i>, 0 = no dominance)</td></tr>
<tr style="height:5px"></tr>
<tr><td style="vertical-align:top"><b style="font-family:times">Δ<sub>r</sub></b></td><td width=30></td><td style="vertical-align:top">Recoverability effect scaled dominance factor (1 = <i>A</i> is completely dominant over <i>B</i>, -1 = <i>B</i> is dominant over <i>A</i>, 0 = no dominance)</td></tr>

<tr style="height:5px"></tr>
<tr><td style="vertical-align:top"><b style="font-family:times">SD in <i>Δ<sub>g</sub></i></b></td><td width=30></td><td style="vertical-align:top">Expected standard deviation in the posterior distribution for <i>Δ<sub>g</sub></i></td></tr>

<tr style="height:5px"></tr>
<tr><td style="vertical-align:top"><b style="font-family:times">SD in <i>Δ<sub>f</sub></i></b></td><td width=30></td><td style="vertical-align:top">Expected standard deviation in the posterior distribution for <i>Δ<sub>f</sub></i></td></tr>

<tr style="height:5px"></tr>
<tr><td style="vertical-align:top"><b style="font-family:times">SD in <i>Δ<sub>r</sub></i></b></td><td width=30></td><td style="vertical-align:top">Expected standard deviation in the posterior distribution for <i>Δ<sub>r</sub></i></td></tr>


</table>
<p>
<b>Interpretation </b> - Quantities giving standard deviations in effect sizes (i.e. SD in <i>a<sub>g</sub></i> /  <i>a<sub>f</sub></i> /  <i>a<sub>r</sub></i>) provide an indication of what size of effect could potentially be identified by the transmission experiment. For example a value "SD in <i>a<sub>g</sub></i>  = 0.2" would mean that statistically significant association can be made in cases in which the <i>AA</i> and <i>BB</i> (or alternatively vaccinated/unvaccinated) individuals differ in their susceptibility by 20% or more. The experiment would likely lack the power to identify smaller effect sizes than this. (Note, this only provides a general guide and the actual statistical power actually depends on stochastic variability inherent in the epidemiological process itself).  
</p> 
 <input class="br" id="number" type="button" value="OK" onclick="paramdefoff();"" style="width:100px;"/>

</div>


</body>
</html>
